cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java

```diff
@@ -99,6 +101,88 @@ public void cleanupTest() throws Exception {
   }
 
   @Test
+  public void testJoiner() throws Exception {
+    testJoinerPlugin(Engine.MAPREDUCE);
+  }
+
+  private void testJoinerPlugin(Engine engine) throws Exception {
+    Schema inputSchema = Schema.recordOf(
+      "testRecord",
+      Schema.Field.of("id", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("name", Schema.of(Schema.Type.STRING))
+    );
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(new ETLStage("source1", MockSource.getPlugin("joinerInput")))
+      .addStage(new ETLStage("tOne", FieldsPrefixTransform.getPlugin("tOne", inputSchema.toString())))
+      .addStage(new ETLStage("tTwo", FieldsPrefixTransform.getPlugin("tTwo", inputSchema.toString())))
+      .addStage(new ETLStage("testJoiner", Join.getPlugin("tOne.tOneid,tTwo.tTwoid:tOne.tOnename,tTwo.tTwoname",
```

Comment:
The notation for join condition specification is difficult to read. Can we use something more readable like  `t1.id=t2.id,t1.name=t2.name` to represent join condition `t1.id = t2.id AND t1.name = t2.name`?